### PR TITLE
chore: update build scripts to produce more reliable caching

### PR DESCRIPTION
**Pull request by moldy530**
# Description
This PR updates the generate script for smart contracts to also lint the code generated because right now everytime you run build it produces unlinted code. This results in the cache being consistently busted for this package but also results in unlinted code being generated locally on each build which is a bit annoying. 

# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing the development configurations across various `turbo.json` files, introducing new tasks, and updating workflow configurations to integrate `OPENAI_API_KEY`. It also modifies package scripts to include linting for generated files and adds the `dotenv-cli` dependency.

### Detailed summary
- Added `dev` tasks in multiple `turbo.json` files with specific dependencies and settings.
- Updated `.github/workflows/on-pull-request.yml` and `.github/workflows/publish-package.yml` to include `OPENAI_API_KEY`.
- Modified `package.json` scripts to include linting for generated files.
- Introduced `dotenv-cli` dependency in `package.json`.
- Updated `yarn.lock` to reflect the new dependency.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
* **#1150** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @moldy530 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: account-kit/react/package.json [hunk 1]
```diff
@@ -38,7 +38,6 @@
     "prebuild": "npx tsx ./inject-version.ts",
     "dev": "storybook dev",
     "build": "yarn clean && yarn build:esm && yarn build:types",
-    "build:dev": "lerna watch --scope='@account-kit/react' --include-dependencies -- lerna run build --scope=@account-kit/react",
```
Comment (moldy530) [inline account-kit/react/package.json:41]: this is now handled correctly by turbo

File: account-kit/smart-contracts/turbo.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "extends": ["//"],
+  "tasks": {
+    "generate": {
+      "env": ["OPENAI_API_KEY"],
```
Comment (moldy530) [inline account-kit/smart-contracts/turbo.json:5]: used when linting the output of the generate

File: site/turbo.json [hunk 3]
```diff
@@ -5,6 +5,12 @@
       "outputs": ["dist/**", "sidebar/**"],
       "dependsOn": ["^build", "^docs:gen"],
       "cache": false
+    },
+    "dev": {
+      "dependsOn": ["^build"],
+      "persistent": true,
+      "cache": false,
+      "interruptible": true
```
Comment (moldy530) [inline site/turbo.json:13]: this doesn't get hot reloaded when packages change, so this one has to be restarted by turbo

**Review by dphilipson — APPROVED** [review]

**Pull request merged**